src/seq_join.rs

```diff
@@ -0,0 +1,361 @@
+use futures::{stream::TryCollect, Future, Stream, TryStreamExt};
+use pin_project::pin_project;
+use std::{
+    collections::VecDeque,
+    future::IntoFuture,
+    iter::Fuse,
+    num::NonZeroUsize,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+/// This helper function might be necessary to convince the compiler that
+/// the return value from [`seq_try_join_all`] implements `Send`.
+/// Use this if you get higher-ranked lifetime errors that mention `std::marker::Send`.
+// pub fn assert_send<'a, F, O>(
+//     fut: F,
+// ) -> F where F: Future<Output = O> + Send + 'a {
+//     fut
+// }
+pub fn assert_send<'a, O>(
+    fut: impl Future<Output = O> + Send + 'a,
+) -> impl Future<Output = O> + Send + 'a {
+    fut
+}
+
+/// Sequentially join futures from an iterator.
+///
+/// This function polls futures from a stream in strict sequence.
+/// If any future blocks, up to `active - 1` futures after it will be polled so
+/// that they make progress.
+///
+/// Unlike [`StreamExt::buffered`], Futures from the stream must resolve in the
+/// same order in which they are produced.
+///
+/// This API accepts an active stream count and a [`Stream`] instance,
+/// from which `Future`s are drawn.
+/// To replicate the effect of [`try_join_all`], an input `Iterator` can be adapted using
+/// [`futures::stream::iter`]; the output can be collected into a [`Vec`] (or any
+/// collection) using [`StreamExt::collect`], as follows:
+///
+/// ```ignore
```

Comment:
These docs are a little out of date.  I still want to take a `Stream` as input here, but working out how to compile that was taking too much energy.